selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceProjects.java

```diff
@@ -11,39 +11,148 @@
     String PROJECT_BY_NAME = "//div[contains(@ng-click, 'projectItem')]/span[text()='%s']";
     String DELETE_PROJECT = "//button/span[text()='Delete']";
     String DELETE_SELECTED_PROJECTS = "//che-button-primary[@che-button-title='Delete']/button";
     String DELETE_IT_PROJECT = "//md-dialog//che-button-primary[@che-button-title='Delete']/button";
     String ADD_NEW_PROJECT_BUTTON = "//che-button-primary[@che-button-title='Add Project']/button";
     String PROJECT_CHECKBOX = "//md-checkbox[contains(@aria-label, 'Project %s')]";
+    String SEARCH_FIELD_XPATH = "//workspace-details-projects//input[@placeholder='Search']";
+    String NOTIFICATION_CONTAINER_ID = "che-notification-container";
+    String DELETE_BUTTON_XPATH = "//che-button-primary[@che-button-title='Delete']";
+    String DELETE_BUTTON_MESSAGE_XPATH = "//che-list-header-delete-button/span";
+    String SELECT_ALL_CHECKBOX_XPATH = "//md-checkbox[@aria-label='Project list']/div";
+    String CHECKBOX_BY_PROJECT_NAME_XPATH_TEMPLATE = "//md-checkbox[@aria-label='Project %s']";
+  }
+
+  public enum BottomButton {
+    SAVE_BUTTON("save-button"),
+    APPLY_BUTTON("apply-button"),
+    CANCEL_BUTTON("cancel-button");
+
+    private final String buttonNameAttribute;
+
+    BottomButton(String buttonNameAttribute) {
+      this.buttonNameAttribute = buttonNameAttribute;
+    }
+
+    public String get() {
+      return this.buttonNameAttribute;
+    }
+  }
+
+  public void clickOnSelectAllCheckbox() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(SELECT_ALL_CHECKBOX_XPATH));
+  }
+
+  public WebElement waitDeleteButton() {
+    return seleniumWebDriverHelper.waitVisibility(By.xpath(DELETE_BUTTON_XPATH));
+  }
+
+  public void waitDeleteButtonDisappearance() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(DELETE_BUTTON_XPATH));
+  }
+
+  public void clickOnDeleteButton() {
+    waitDeleteButton().click();
+  }
+
+  public void waitDeleteButtonMessage(String expectedMessage) {
+    seleniumWebDriverHelper.waitTextContains(
+        By.xpath(DELETE_BUTTON_MESSAGE_XPATH), expectedMessage);
+  }
+
+  public void waitDeleteButtonMessageDisappearance() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(DELETE_BUTTON_MESSAGE_XPATH));
+  }
+
+  public void waitDeleteButtonDisabling() {
+    waitButtonState(By.xpath(DELETE_BUTTON_XPATH), false);
+  }
+
+  public void waitDeleteButtonEnabling() {
+    waitButtonState(By.xpath(DELETE_BUTTON_XPATH), true);
+  }
+
+  public void typeToSearchField(String text) {
+    seleniumWebDriverHelper.setValue(waitSearchField(), text);
+  }
+
+  public void waitNotification(String expectedText) {
+    seleniumWebDriverHelper.waitTextContains(By.id(NOTIFICATION_CONTAINER_ID), expectedText);
+  }
+
+  public WebElement waitBottomButton(BottomButton button) {
+    return seleniumWebDriverHelper.waitVisibility(By.name(button.get()));
+  }
+
+  public void waitBottomButtonInvisibility(BottomButton... bottomButtons) {
```

Comment:
`Button` > `Buttons`

Reply:
No, this is exactly invisibility only of the bottom buttons

Reply:
Or do you mean that it clear from method signature?

Reply:
Yes, exactly. There couldn't be any other types of buttons, but *BottomButton* due to list of parameters. So method name could be reduced to simply `waitInvisiibility(BottomButton... bottomButtons)` without loosing of meaning, IMHO.

Reply:
Good catch, thank you